instant-xml/tests/inline.rs

```diff
@@ -0,0 +1,39 @@
+use instant_xml::{from_str, to_string, FromXml, ToXml};
+
+#[derive(Debug, Eq, FromXml, PartialEq, ToXml)]
+struct Wrapper {
+    inline: Inline,
+}
+
+#[derive(Debug, Eq, FromXml, PartialEq, ToXml)]
+#[xml(inline)]
+struct Inline {
+    foo: Foo,
+    bar: Bar,
+}
```

Comment:
Does this also support the following?
```rust
struct Wrapper {
    inline: Vec<Inline>,
}

#[derive(Debug, Eq, FromXml, PartialEq, ToXml)]
#[xml(inline)]
enum Inline {
    Foo(Foo),
    Bar(Bar),
}
```
Wondering which data structure is better. Imagine Vec to be some non-allocating collection.

Reply:
Yeah, I think that should work? Don't have a use case for it yet. The case in instant-epp where we need this is where we're explicitly combining two extensions so in that case IMO being explicit about the two fields is nicer.

Reply:
My use case was also part of an idea for building your own set of supported extensions, e.g. like this:
```rust
// In lib
struct Wrapper<T = Inline> {
    inline: Vec<T>,
}

#[derive(Debug, Eq, FromXml, PartialEq, ToXml)]
enum Inline {
    Foo(Foo),
    Bar(Bar),
}

// In your own code you can then do something like this
type Response = Wrapper<AllExtensionsEnabledResponse>;
enum AllExtensionsEnabledResponse {
    Foo(Foo),
    Bar(Bar),
    Baz(Baz)
}
```
Which one is more ergonomic probably depends on the exact case (e.g. if all fields are optional or not)
Downside would be one more generic inside. Could hurt compile times.

Reply:
Sure, but for now it seems like the current approach will do? At least I'm not sure what bearing this thread has on this PR.